lib/icingadb/icingadb-objects.cpp

```diff
@@ -941,12 +941,31 @@ void IcingaDB::InsertObjectDependencies(const ConfigObject::Ptr& object, const S
 					values = new Dictionary({{"value", kv.second}});
 				}
 
-				{
+				for (const char *attr : {"value", "set_if"}) {
 					Value value;
 
-					// JsonEncode() the value if it's set.
-					if (values->Get("value", &value)) {
-						values->Set("value", JsonEncode(value));
+					// Stringify if set.
+					if (values->Get(attr, &value)) {
+						switch (value.GetType()) {
```

Comment:
Why would it be a problem to just use `JsonEncode` for both `value` and `set_if` just as it was done before? This would avoid this whole switch block and duplication of type checks with `JsonEncode`.

With the current state of the PR, the value can be anything, maybe it's JSON, maybe it's not. This doesn't seem very helpful.

Reply:
1. As far as I remember we've decided to use K.I.S.S. strings.
2. The concept controls your implementation details, not vice-versa.

Reply:
Regarding 2.: if the implementation gets this messy, maybe it's worth reconsidering the concept.

Apart from that, the PR does not even state what the purpose of this PR is. Can you please document this?